miden-lib/asm/miden/contracts/wallets/basic.masm

```diff
@@ -73,3 +73,180 @@ export.send_asset.1
     # prepare the stack for return - stack has 6 elements too many
     movupw.3 dropw swap drop swap drop
 end
+
+#! Creates a note without assets.
+#!
+#! Inputs stack: [tag, aux, note_type, RECIPIENT, ...]
+#! Outputs: [note_idx, 0, 0, EMPTY_WORD, ...]
+#!
+#! - tag is the tag to be included in the note.
+#! - aux is the auxiliary data to be included in the note.
+#! - note_type is the note's storage type
+#! - RECIPIENT is the recipient of the note, i.e.,
+#!   hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash)
+export.cteate_note
+    # This procedure is written to be executed with `call`. It means that has to be padded to the 
+    # length of 16 to preserve the stack state after the call. 
+    push.0 movdn.7 padw movdnw.2 padw movdnw.3
+    # => [tag, aux, note_type, RECIPIENT, 0, EMPTY_WORD, EMPTY_WORD, ...]
```

Comment:
I'm not quite following what this is doing. In general the padding should be done by the caller - i.e., the signature of this procedure should be:
```
#! Inputs:  [tag, aux, note_type, RECIPIENT, PAD(9)]
#! Outputs: [note_idx, PAD(15)]
```
Basically, the procedure takes 16 elements as input (though the last 9 elements are irrelevant) and returns 16 elements (though, the lsat 15 elements are irrelevant). But the procedure makes no guarantees as to the values of the padded elements - so, the caller cannot assume anything about them.

It will be the responsibility of the caller to pad the stack appropriately before the `call` instruction and then to remove the padding after the call instruction.

Reply:
I see, now I understand what did you mean. I'll update the stack states before and after the `call`s